UniAtHome/UniAtHome.WebAPI/Controllers/CourseController.cs

```diff
@@ -32,38 +37,18 @@ public async Task<IActionResult> GetCourseById(int id)
 
         // POST: api/Course
         [HttpPost]
-        public async Task<IActionResult> CreateCourse([FromBody] CourseDTO course)
+        [Authorize]
+        public async Task<IActionResult> CreateCourse([FromBody] CreateCourseRequest request)
         {
-            if (course != null && ModelState.IsValid)
+            if (request != null && ModelState.IsValid)
             {
-                await this.courseService.AddCourseAsync(course);
+                var courseDto = mapper.Map<CreateCourseRequest, CourseDTO>(request);
+                courseDto.TeacherEmail = User.Identity.Name;
+                await courseService.AddCourseAsync(courseDto);
                 return Ok();
             }
 
             return BadRequest();
         }
-
-        [HttpPut("{id}")]
-        public void UpdateCourse(int id, [FromBody] string value)
-        {
-        }
-
-        [HttpDelete("{id}")]
-        public void Delete(int id)
-        {
-        }
-
-        // GET: api/Course/byname?name=AVPZ
-        [HttpGet]
-        public async Task<IActionResult> GetCourseByName(string name)
-        {
-            var courses = await this.courseService.GetCoursesByNameAsync(name);
-            if (courses != null)
-            {
-                return Ok(courses);
-            }
-
-            return BadRequest();
-        }
```

Comment:
Why did you delete this method and didn't create any alternative one? It was fully operation at least until it was changed.

Reply:
I will create an alternative in the next PR. I will provide universal search endpoint.